lib/offline_sort/merger.rb

```diff
@@ -0,0 +1,55 @@
+require 'forwardable'
+
+module OfflineSort
+  class Merger
+    extend Forwardable
+    include Enumerable
+
+    attr_reader :sorted_chunks, :sort_by
+
+    def initialize(sorted_chunks, sort_by)
+      @sorted_chunks = sorted_chunks
+      @sort_by = sort_by
+    end
+
+    def_delegators :enumerator, :each
+
+    private
+
+    def enumerator
+      pq = []
+      chunk_enumerators =
+        sorted_chunks.each(&:open).each(&:rewind).map(&:each)
```

Comment:
Totally fine to leave as-is, just chipping in my two cents here!

It always worries me a little bit to rely on the return value of `#each`, since I think it only returns the original enumerable for `Array`s. When I run into something like this I'll usually reach for `#tap`, i.e.
```ruby
sorted_chunks.tap do |chunks|
  chunks.each(&:open)
  chunks.each(&:rewind)
end.map(&:each)
```
🤷